### PR TITLE
feature/205-profile-update

### PR DESCRIPTION
**Pull request by 3nin6**
This PR aims to resolve issue #205 

Args `--name --server --username` are no longer `required=True`. In order to prevent this from allowing users to create profiles with missing information a catch was added to create_profile.

### CODE REVIEW

**Comment by 3nin6** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/code42cli/profile.py [hunk 1]
```diff
@@ -102,7 +102,8 @@ def switch_default_profile(profile_name):
 def create_profile(name, server, username, ignore_ssl_errors):
     if profile_exists(name):
         raise Code42CLIError("A profile named '{}' already exists.".format(name))
-
+    if None in {name, server, username}:
```
Comment (unparalleled-js) [inline src/code42cli/profile.py:105]: This causes the `[required]` documentation help text to go away from the `create` command, where these are actually required. An alternative approach that will allow the help text to remain accurate would be to use parameterized methods for creating the `click.options`.

Example:

```
def usename_option(required=False):
  return click.option(
    "-u", "--username", required=required, help="The username of the Code42 API user.",
```

Then, in the update function, use it like this:

```
@profile.command()
...
@username_option()
def update()...
```

And in the create method,

```
@profile.command()
...
@username_option(required=True)
def create()...
```

**Comment by unparalleled-js** [comment]
Thank you so much for this contribution!

A couple of things that will be needed before merging are tests and a CHANGELOG entry. Let me know if you any help with those.

I also added a comment on an alternative way to achieve this without sacrificing the parts of the help text that specify which args are required (for the create command).

**Comment by 3nin6** [comment]
Thank you for the feedback, I'll look at making changes tonight to improve the PR.

**Comment by 3nin6** [comment]
Hi @unparalleled-js , I have made the changes for the functionality and added a preliminary changelog entry based off of others PRs I checked. I am having trouble getting a test to check this without making style checks fail permanently - they make the changes every time I run tox.

first pass at a test for this is a clone of the updates_existing_profile test it but without passing the `-n` option:

`def test_update_profile_updates_default_profile(
    runner, mock_cliprofile_namespace, user_agreement, valid_connection, profile
):
    name = "foo"
    profile.name = name
    mock_cliprofile_namespace.get_profile.return_value = profile
    runner.invoke(
        cli,
        [
            "profile",
            "update",
            "-s",
            "bar",
            "-u",
            "baz",
            "--disable-ssl-errors",
        ],
    )
    mock_cliprofile_namespace.update_profile.assert_called_once_with(
        name, "bar", "baz", True
    )
`

**Comment by unparalleled-js** [comment]
> Hi @unparalleled-js , I have made the changes for the functionality and added a preliminary changelog entry based off of others PRs I checked. I am having trouble getting a test to check this without making style checks fail permanently - they make the changes every time I run tox.
> 
> first pass at a test for this is a clone of the updates_existing_profile test it but without passing the `-n` option:
> 
> ```
> def test_update_profile_updates_default_profile(
>     runner, mock_cliprofile_namespace, user_agreement, valid_connection, profile
> ):
>     name = "foo"
>     profile.name = name
>     mock_cliprofile_namespace.get_profile.return_value = profile
>     runner.invoke(
>         cli,
>         [
>             "profile",
>             "update",
>             "-s",
>             "bar",
>             "-u",
>             "baz",
>             "--disable-ssl-errors",
>         ],
>     )
>     mock_cliprofile_namespace.update_profile.assert_called_once_with(
>         name, "bar", "baz", True
>     )
> ```

Correct me I am not understand correctly, but

If you run `tox -e style`, it will correct any style issues.

The test you provided here looks like the right idea.  We could add another test as well that checks that `code42 profile update -u <username>` works `test_update_profile_updates_username_alone()`

**Comment by 3nin6** [comment]
I have committed the tests following running `tox` and also just `tox -e style` for review. I received the same failure+correction diff each time I ran it.

**Comment by alanag13** [comment]
> If you run `tox -e style`, it will correct any style issues.

flake8 issues get checked with `tox -e style` as well, and those need to be corrected manually.
In your case, the output shows:

```
tests/cmds/test_profile.py:264:84: E231 missing whitespace after ','
tests/cmds/test_profile.py:278:71: E231 missing whitespace after ','
```

It looks like in both cases this is caused by an unnecessary trailing comma at the end of a list.

File: tests/cmds/test_profile.py [hunk 2]
```diff
@@ -254,6 +254,34 @@ def test_update_profile_updates_existing_profile(
...
 
 
+def test_update_profile_updates_default_profile(
+    runner, mock_cliprofile_namespace, user_agreement, valid_connection, profile
+):
+    name = "foo"
+    profile.name = name
+    mock_cliprofile_namespace.get_profile.return_value = profile
+    runner.invoke(
+        cli, ["profile", "update", "-s", "bar", "-u", "baz", "--disable-ssl-errors",],
```
Comment (alanag13) [inline tests/cmds/test_profile.py:264]: here and only line 278 removing the last `,` before the closing `]` should resolve the style errors.

Comment (3nin6) [reply tests/cmds/test_profile.py]: Thank you for pointing that out, I didn't realize the style corrections weren't fully automated.

**Review by unparalleled-js — APPROVED** [review]

File: tests/cmds/test_profile.py (hunk 2 above)
Comment (unparalleled-js) [reply tests/cmds/test_profile.py]: Thanks Alan

**Pull request merged**